lib/ansible/modules/cloud/amazon/route53_health_check.py

```diff
@@ -128,9 +141,18 @@
     state: absent
     fqdn: host1.example.com
 
+- route53_health_check:
+    state: present
+    type: CALCULATED
+    child_health_checks:
+      - "{{ my_health_check.health_check.id }}"
+      - "{{ another_health_check.health_check.id }}"
+    health_threshold: 2
+  register: my_health_check
 '''
 
 import uuid
+import ipaddress
```

Comment:
I have isolated the bug fix in this commit: https://github.com/ansible/ansible/pull/24112/commits/1e9da96a4a8ab7960c6af817cdd265edc47c03c2

IPv6 addresses returned by Route53 does not match compressed or exploded python format and thus comparison is incorrect. I use the `ipaddress` module to always compare IP address as compressed so they can match if they are equal.

For example, `:0000:0000:` becomes:
- `::` compressed
- `:0000:0000:` exploded
- `:0:0` in Route53 responses